omc3/utils/iotools.py

```diff
@@ -58,7 +58,7 @@ def copy_item(src_item, dest):
         LOG.error(f"Could not copy item because of IOError. Item: '{src_item}'")
 
 
-def glob_regex(path: Path, pattern: str) -> "filter object":
+def glob_regex(path: Path, pattern: str) -> Sequence[str]:
```

Comment:
I don't think this is correct. See e.g. https://stackoverflow.com/q/33174276 or test it yourself.
On the other hand, this also regularly is marked by my linter/syntax checker as an error...

Reply:
Indeed, this would be a correct type hint if we cast to `list` before returning. I'll just put a `filter` class as hint.

Reply:
I'd also be ok with rewriting:

```python
def glob_regex(path: Path, pattern: str) -> Iterator[str]:
""" Do a glob on the given `path` based on the regular expression `pattern`.
    Returns only the matching filenames (as strings).
    Args:
        path (Path): Folder path to look in.
        pattern (str): Pattern to match.
    """
    regex = re.compile(pattern)
    names = (p.name for p in path.glob("*")
    return (name for name in names if regex.match(name))
 ```